File: seed/php-sdk/pagination/no-custom-config/src/Types/UsernamePage.php
<?php

namespace Seed\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;
use Seed\Core\Types\ArrayType;

class UsernamePage extends JsonSerializableType
{
    /**
     * @var ?string $after
     */
    #[JsonProperty('after')]
    public ?string $after;

    /**
     * @var array<string> $data
     */
    #[JsonProperty('data'), ArrayType(['string'])]
    public array $data;

    /**
     * @param array{
     *   after?: ?string,
     *   data: array<string>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->after = $values['after'] ?? null;
        $this->data = $values['data'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: seed/csharp-sdk/pagination/custom-pager-with-exception-handler/src/SeedPagination/Types/UsernamePage.cs
using System.Text.Json.Serialization;
using SeedPagination.Core;

namespace SeedPagination;

public record UsernamePage
{
    [JsonPropertyName("after")]
    public string? After { get; set; }

    [JsonPropertyName("data")]
    public IEnumerable<string> Data { get; set; } = new List<string>();

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
